x/fungible/migrations/v4/migrate.go

```diff
@@ -0,0 +1,79 @@
+package v4
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethermint "github.com/zeta-chain/ethermint/types"
+
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/x/fungible/types"
+)
+
+type fungibleKeeper interface {
+	GetAllForeignCoins(ctx sdk.Context) (list []types.ForeignCoins)
+	QuerySystemContractGasCoinZRC20(ctx sdk.Context, chainid *big.Int) (ethcommon.Address, error)
+	ZRC20BalanceOf(ctx sdk.Context, zrc20Address, owner ethcommon.Address) (*big.Int, error)
+	CallZRC20Burn(
+		ctx sdk.Context,
+		sender ethcommon.Address,
+		zrc20address ethcommon.Address,
+		amount *big.Int,
+		noEthereumTxEvent bool,
+	) error
+}
+
+// MigrateStore migrates the store from consensus version 3 to 4.
+// It burns the SUI gas ZRC20 from the stability pool address.
+func MigrateStore(ctx sdk.Context, fungibleKeeper fungibleKeeper) error {
+	chainID, err := ethermint.ParseChainID(ctx.ChainID())
+	if err != nil {
+		// It's fine to return nil here and not try to execute the migration at all if the parsing fails
+		ctx.Logger().Error("failed to parse chain ID", "chain_id", ctx.ChainID(), "error", err)
```

Comment:
@swift1337 
Answering https://github.com/zeta-chain/node/pull/4016#discussion_r2209793782

Keeping the error log here is fine, this would be executed only once at the upgrade height , we are not returning an error so it is not redundant in this case